gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/TaskDtoBuilder.java

```diff
@@ -42,50 +41,70 @@ public TaskDto getTaskDto(OffsetDateTime timestamp) {
     }
 
     public List<TaskDto> getListTasksDto(LocalDate businessDate) {
-        List<TaskDto> listTasks = new ArrayList<>();
         ZoneId zone = ZoneId.of(properties.getProcess().getTimezone());
         LocalDateTime businessDateStartTime = businessDate.atTime(0, 30);
+        LocalDateTime businessDateEndTime = businessDate.atTime(23, 30);
         ZoneOffset zoneOffSet = zone.getRules().getOffset(businessDateStartTime);
-        OffsetDateTime timestamp = businessDateStartTime.atOffset(zoneOffSet);
-        while (timestamp.getDayOfMonth() == businessDate.getDayOfMonth()) {
-            listTasks.add(getTaskDto(timestamp));
-            timestamp = timestamp.plusHours(1).atZoneSameInstant(zone).toOffsetDateTime();
+        OffsetDateTime startTimestamp = businessDateStartTime.atOffset(zoneOffSet);
+        OffsetDateTime endTimestamp = businessDateEndTime.atOffset(zoneOffSet);
+        Set<Task> tasks = taskRepository.findAllByTimestampBetweenForBusinessDayView(startTimestamp, endTimestamp);
+        Map<OffsetDateTime, TaskDto> taskMap = new HashMap<>();
+        for (OffsetDateTime loopTimestamp = startTimestamp;
+             !loopTimestamp.isAfter(endTimestamp);
+             loopTimestamp = loopTimestamp.plusHours(1).atZoneSameInstant(zone).toOffsetDateTime()
+        ) {
+            taskMap.put(loopTimestamp, getEmptyTask(loopTimestamp));
         }
-        return listTasks;
+        tasks.stream().map(t -> createDtoFromEntityWithOrWithoutEvents(t, false)).forEach(dto -> taskMap.put(dto.getTimestamp(), dto));
+
+        return taskMap.values().stream().toList();
     }
 
     public List<TaskDto> getListRunningTasksDto() {
-        return taskRepository.findAllRunningAndPending().stream().map(this::createDtoFromEntity).collect(Collectors.toUnmodifiableList());
+        return taskRepository.findAllRunningAndPending().stream().map(t -> createDtoFromEntityWithOrWithoutEvents(t, false)).toList();
```

Comment:
Previously, the list was unmodifiable. Was there a particular reason? If yes, then we should reasonably keep the list unmodifiable in the new version of the code. Else, no problem.

Reply:
no particular reason